### PR TITLE
[bug](distinct agg) fix distinct streaming agg not output all data

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
in distinct agg, will insert data into out_block with selector, but we not push out_block into data queue every times,
until out_block rows >= batch_size.
and _stop_emplace_flag = true, means have no need to emplace value into hash table, it should be streaming.
but maybe have some data in out_block, need output those firstly.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by hello-stephen — APPROVED** [review]

File: be/src/vec/exec/distinct_vaggregation_node.cpp [hunk 1]
```diff
@@ -64,14 +65,25 @@ Status DistinctAggregationNode::_distinct_pre_agg_with_serialized_key(
...
+            key_columns[idx]->append_data_by_selector(dst, _distinct_row);
+        };
         for (int i = 0; i < key_size; ++i) {
             auto output_column = out_block->get_by_position(i).column;
-            if (_stop_emplace_flag) { // swap the column directly, to solve Check failed: d.column->use_count() == 1 (2 vs. 1)
+            if (_stop_emplace_flag && !out_block->empty()) {
+                // when out_block row >= batch_size, push it to data_queue, so when _stop_emplace_flag = true, maybe have some data in block
+                // need output those data firstly
+                for (int i = 0; i < rows; ++i) {
+                    _distinct_row.push_back(i);
```
Comment (BiteTheDDDDt) [inline be/src/vec/exec/distinct_vaggregation_node.cpp:78]: Will _distinct_row be pushed back multiple times?

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**